assets/wizards/readerRevenue/views/stripe-setup/index.js

```diff
@@ -103,6 +105,115 @@ export const StripeKeysSettings = () => {
 	);
 };
 
+const WebhooksList = ( { webhooks, onUpdate } ) => {
+	const { wizardApiFetch } = useDispatch( Wizard.STORE_NAMESPACE );
+	const isLoading = useSelect( select => select( Wizard.STORE_NAMESPACE ).isQuietLoading() );
+
+	const handleChange = ( webhook, method, payload ) => {
+		const args = {
+			path: `/newspack/v1/stripe/webhook/${ webhook.id }`,
+			method,
+			data: payload,
+			isQuietFetch: true,
+		};
+		if ( payload ) {
+			args.data = payload;
+		}
+		wizardApiFetch( args ).then( onUpdate( method ) );
+	};
+
+	return (
+		<ul>
+			{ webhooks.map( webhook => {
+				return (
+					<li key={ webhook.id }>
+						<span>
+							<ToggleControl
+								checked={ webhook.status === 'enabled' }
+								disabled={ isLoading }
+								onChange={ () =>
+									handleChange( webhook, 'POST', {
+										status: webhook.status === 'enabled' ? 'disabled' : 'enabled',
+									} )
+								}
+							/>
+							<code>{ webhook.url }</code>
+						</span>
+
+						<Button
+							isDestructive
+							icon={ trash }
+							disabled={ isLoading }
+							onClick={ () => handleChange( webhook, 'DELETE' ) }
+						/>
+					</li>
+				);
+			} ) }
+		</ul>
+	);
+};
+
+const StripeWebhooksSettings = () => {
+	const { stripe_data: data = {} } = Wizard.useWizardData( 'reader-revenue' );
+	const { updateWizardSettings } = useDispatch( Wizard.STORE_NAMESPACE );
+	const webhooksList = data.webhooks_list || [];
+
+	const [ thisSiteWebhooks, otherWebhooks ] = webhooksList.reduce(
+		( acc, webhook ) => {
+			if ( webhook.matches_url ) {
+				acc[ 0 ].push( webhook );
+			} else {
+				acc[ 1 ].push( webhook );
+			}
+			return acc;
+		},
+		[ [], [] ]
+	);
+	const activeThisSiteWebhooks = thisSiteWebhooks.filter( webhook => webhook.status === 'enabled' );
+
+	const handleWebhooksListUpdate = method => payload => {
+		let newList = webhooksList;
+		switch ( method ) {
+			case 'POST':
+				newList = webhooksList.map( webhook => ( payload.id === webhook.id ? payload : webhook ) );
+				break;
+			case 'DELETE':
+				newList = webhooksList.filter( webhook => payload.id !== webhook.id );
+				break;
+		}
+		updateWizardSettings( {
+			slug: READER_REVENUE_WIZARD_SLUG,
+			path: [ 'stripe_data', 'webhooks_list' ],
+			value: newList,
+		} );
+	};
+	return (
+		<div className="newspack-payment-setup-screen__webhooks">
+			{ activeThisSiteWebhooks.length > 1 && (
+				<Notice
+					isError
+					noticeText={ __(
+						'There are too many webhoooks for this site. Please delete or disable the extra ones.',
+						'newspack'
+					) }
+				/>
+			) }
+			{ activeThisSiteWebhooks.length === 0 && (
+				<Notice
+					isError
+					noticeText={ __( 'There are no active webhoooks for this site.', 'newspack' ) }
+				/>
+			) }
+			<WebhooksList webhooks={ thisSiteWebhooks } onUpdate={ handleWebhooksListUpdate } />
```

Comment:
This initial view was confusing to me for a bit:

<img width="1075" alt="Screen Shot 2022-11-30 at 4 55 58 PM" src="https://user-images.githubusercontent.com/2230142/204933220-7e7ae63b-b862-4840-a9ee-e7a514aa7e68.png">

The "Webhooks not connected to this site." accordion heading at first looked to me like a message related to the webhook for my site because of the plain styling. It also wasn't immediately clear to me that I could click on that heading to display other webhooks.

Lastly, it's a little confusing to me that the toggle switches trigger an immediate change, with the "Save Settings" button being displayed right below them. It feels unclear at first whether I'll need to click that button to persist any changes I make.

Suggestions:

* Provide a more prominent heading for each list: maybe "Webhooks for this site" and "Other webhooks"
* I think the Accordion component will be really useful going forward, but the styling seems off to me. Can we increase the font-size and maybe apply some styling to the Accordion component's heading to make it look more like a heading inside a button? I can imagine it more visually resembling one of the rows in the expanded view, or perhaps wrap everything in a container panel like the [Material accordion](https://m1.material.io/components/expansion-panels.html).
* Perhaps move this whole section out into a page of its own? Since it's a relatively advanced feature, maybe it doesn't have to live alongside Stripe Settings in the main wizard nav, but can be accessed via a button or action card clicked from the Stripe Settings page, kind of like the GAM configuration settings in the Advertising wizard.
* Consider having a link out to the Stripe dashboard URL (https://dashboard.stripe.com/) to make it clearer that you're modifying data that lives in that external dashboard. If it's possible to intuit the Stripe dashboard page for each webhook from the webhook data, it might be useful to link out each webhook to that page in Stripe.

Reply:
@thomasguillot I see you already gave a 👍 on the design, so I'd defer to you as to whether we should consider any of my above suggestions.

Reply:
If we move this to its own page, I'm not sure we actually need an accordion. 

<img width="1091" alt="Screenshot 2022-12-06 at 12 29 51" src="https://user-images.githubusercontent.com/177929/205913133-1fa6789b-c585-4f68-8d3a-56e447698a3c.png">

Clicking on "Edit Webhooks" would then take the user to a new screen:

![webhooks i3](https://user-images.githubusercontent.com/177929/205913265-ddb7bb3b-a9bb-459c-8ba3-3440fe3bd4b3.png)

Reply:
>  it's a little confusing to me that the toggle switches trigger an immediate change, with the "Save Settings" button being displayed right below them.

I agree with this concern, and moving to a separate page sounds like a good solution. I don't like the proposed placement of the "Edit Webhooks" link, though. It feels out of place there, as the secondary action to saving the settings. An action card seems like the idiom we use in these cases:

<img width="852" alt="image" src="https://user-images.githubusercontent.com/7383192/206145929-fd45c549-efb2-4149-b5ec-1397254a69e6.png">

Reply:
I'm not a fan of the `ActionCard` underneath the "Save Settings" `Button` but you're correct, it's better than a secondary `Button`.

Reply:
>  it's a little confusing to me that the toggle switches trigger an immediate change, with the "Save Settings" button being displayed right below them.

I second this concern, and the suggestion to move it to a separate page. I don't like the proposed placement of the "Edit Webhooks" link, though. It feels out of place as a secondary action to settings saving. In other places we use the action card to link to other pieces of UI:

<img width="852" alt="image" src="https://user-images.githubusercontent.com/7383192/206148908-5c368674-2068-46ca-9c27-c6891733739d.png">

Reply:
> I think the Accordion component will be really useful going forward, but the styling seems off to me. 

> If we move this to its own page, I'm not sure we actually need an accordion.

I think we do – the "other webhooks" section is not of interest 90% of the time. This UI is primarily for debugging the webhooks connected to the site. I agree with Derrick that the styling is a bit off – another solution would be to change the title content to indicate user action – e.g. "Click here to view webhooks not connected to the site". 

Also, the "Reset Webhooks" is an action that influences the connected webhook, not the other ones, so it should be placed accordingly – below the first list.

Reply:
About the Accordion's styling – WDYT about something akin to the action card?

<img width="835" alt="image" src="https://user-images.githubusercontent.com/7383192/206158271-54e11628-2630-4fff-809f-48af1d08d6ce.png">

Reply:
![webhooks i4](https://user-images.githubusercontent.com/177929/206187104-5047bdc6-1ce0-4585-8629-5f463db20479.png)

how about something like this then.

Reply:
I see. So we could move the reset button at the bottom, next to the "Back" button (similar to my previous screen)?

Reply:
It should be clear from the UI that the reset is about the "connected" section – otherwise the user might think this button resets _all_ of their webhooks.

Reply:
Got it, thanks @adekbadek. Let's go ahead with this:

![site](https://user-images.githubusercontent.com/177929/207839427-e81c3b22-0d9c-46b4-bd30-ee40ea02f478.png)

I based the "Reset" button on a small secondary button.

Reply:
a433814d61d38cd884c8466f64f2c3eefbabbf91 implements the latest changes. This PR is again ready for review, @dkoo.